axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceTermServiceImpl.java

```diff
@@ -1046,11 +1046,12 @@ public void setParentFields(
       if (moveLine.getPartner() != null) {
         invoiceTerm.setPartner(moveLine.getPartner());
       } else {
-        invoiceTerm.setPartner(move.getPartner());
+        if (move != null)
+          invoiceTerm.setPartner(move.getPartner());
       }
     }
 
-    if (moveLine != null && invoiceTerm.getOriginDate() == null) {
+    if (move != null && invoiceTerm != null && invoiceTerm.getOriginDate() == null) {
```

Comment:
I think you need to keep the test on moveLine because it's a functional test more than a technical one.